zcash_client_backend/src/scanning.rs

```diff
@@ -736,6 +745,12 @@ where
             },
             Ok,
         )?;
+    let sapling_final_tree_size = sapling_commitment_tree_size
+        + block
+            .vtx
+            .iter()
+            .map(|tx| u32::try_from(tx.outputs.len()).unwrap())
+            .sum::<u32>();
```

Comment:
Nit: if we're now going to compute this for every block (instead of just for blocks where we don't have prior block metadata), we should instead extract `sapling_output_count` from inside the `sapling_commitment_tree_size` definition, and use the result both there and here.